client/securedrop_client/sdk/__init__.py

```diff
@@ -592,30 +598,24 @@ def download_submission(
         if path and os.path.exists(path) and not os.path.isdir(path):
             raise BaseError("Please provide a valid directory to save.")
 
-        data, status_code, headers = self._send_json_request(
+        response = self._send_json_request(
             method,
             path_query,
+            stream=True,
             headers=self.req_headers,
             timeout=timeout or self.default_download_timeout,
         )
 
-        if status_code == 404:
-            raise WrongUUIDError(f"Missing submission {submission.uuid}")
-
-        if not self.proxy:
-            # This is where we will save our downloaded file
-            filepath = os.path.join(path, submission.filename)
-            with open(filepath, "wb") as fobj:
-                for chunk in data.iter_content(chunk_size=1024):  # Getting 1024 in each chunk
-                    if chunk:
-                        fobj.write(chunk)
+        if isinstance(response, JSONResponse):
+            if response.status == 404:
+                raise WrongUUIDError(f"Missing submission {submission.uuid}")
+            else:
+                raise BaseError(f"Unknown error, status code: {response.status}")
 
-        else:
-            filepath = os.path.join(
-                "/home/user/QubesIncoming/", self.proxy_vm_name, data["filename"]
-            )
+        filepath = os.path.join(path, submission.filename)
```

Comment:
We can't use the default `path = ""` here; we need to construct a path from `Controller.home`.  Thanks to AppArmor for making this a hard failure:

```
May  9 13:12:10 sd-app 2024-05-09 13:12:10,586 - securedrop_client.queue:244(process) ERROR: Skipping job
May  9 13:12:10 sd-app kernel: [  148.768819] kauditd_printk_skb: 390 callbacks suppressed
May  9 13:12:10 sd-app kernel: [  148.768826] audit: type=1400 audit(1715285530.581:457): apparmor="DENIED" operation="mknod" class="file" profile="/usr/bin/securedrop-client" name="/opt/venvs/securedrop-client/1-scholastic_runner-up-msg.gpg" pid=1151 comm="QThread" requested_mask="c" denied_mask="c" fsuid=1000 ouid=1000
May  9 13:12:10 sd-app 2024-05-09 13:12:10,594 - securedrop_client.logic:874(on_message_download_failure) ERROR: Could not emit message_download_failed
```

Reply:
`path` is always empty (only specified in tests), so we mostly have to pick a place where to store them...using QubesIncoming isn't great, how about ~/Downloads? Let me put together a patch...

Reply:
https://github.com/freedomofpress/securedrop-client/pull/1718/commits/8e77dd8367eef67944ad20f276f5be706a39bd3b - but I'm not super satisfied with it, I outlined a possibly better solution in the commit message.

Reply:
Thanks, @legoktm.  I've amended this in in 8203a8e1bfb2aa31659c972b109b101d6f0ee4b4 to cover `API.download_reply()` too.

I don't like `~/Downloads` any *less* than I like `~/QubesIncoming`, since I take both to be directories owned by the user that applications shouldn't use for transient files under their control.  :-)  But I'll open a follow-up ticket for that change, which will also solve the problem you describe in <https://github.com/freedomofpress/securedrop-client/pull/1718#discussion_r1596167452>: the branch (code-path) we test here isn't actually the branch we care about.